concrete/src/Entity/User/GroupRoleChange.php

```diff
@@ -16,157 +20,151 @@
+     * @var int|null NULL if not yet flushed to the database
      */
     protected $id;
 
     /**
-     * @var int
      * @ORM\OneToMany(targetEntity="\Concrete\Core\Entity\Notification\GroupRoleChangeNotification", mappedBy="signup", cascade={"remove"}),
+     *
+     * @var \Doctrine\Common\Collections\Collection
      */
     protected $notifications;
 
     /**
-     * @var int
      * @ORM\Column(type="integer", options={"unsigned":true})
+     *
+     * @var int
      */
     protected $gID;
 
     /**
-     * @var int
      * @ORM\Column(type="integer", options={"unsigned":true})
+     *
+     * @var int
      */
     protected $grID;
 
     /**
-     * @var User
      * @ORM\ManyToOne(targetEntity="\Concrete\Core\Entity\User\User")
      * @ORM\JoinColumn(name="uID", referencedColumnName="uID", onDelete="SET NULL")
+     *
+     * @var \Concrete\Core\Entity\User\User|null
      */
     protected $user;
 
     /**
-     * @var \DateTime
      * @ORM\Column(type="datetime")
+     *
+     * @var \DateTime
      */
-    protected $requested = null;
+    protected $requested;
 
     /**
-     * GroupSignupRequest constructor.
-     * @param Group $group
-     * @param \Concrete\Core\User\User $user
-     * @param GroupRole $role
-     * @throws \Exception
+     * @param \Concrete\Core\User\Group\Group|null $group
+     * @param \Concrete\Core\User\User|null $user
+     * @param \Concrete\Core\User\Group\GroupRole|null $role
      */
     public function __construct($group = null, $user = null, $role = null)
     {
+        $this->notifications = new ArrayCollection();
         if ($group instanceof Group) {
             $this->gID = $group->getGroupID();
         }
-
         if ($user instanceof \Concrete\Core\User\User) {
-            if ($user->getUserInfoObject() instanceof UserInfo) {
-                $this->user = $user->getUserInfoObject()->getEntityObject();
+            $userInfo = $user->getUserInfoObject();
+            if ($userInfo instanceof UserInfo) {
+                $this->user = $userInfo->getEntityObject();
             }
         }
-
         if ($role instanceof GroupRole) {
             $this->grID = $role->getId();
         }
-
-        $this->requested = new \DateTime();
+        $this->requested = new DateTime();
     }
 
     /**
-     * @return Group
+     * @return \Concrete\Core\User\Group\Group|null
      */
     public function getGroup()
     {
         return Group::getByID($this->getGID());
     }
 
-    public function getRole() {
+    /**
+     * @return \Concrete\Core\User\Group\GroupRole|false
+     */
+    public function getRole()
+    {
         return GroupRole::getByID($this->getGrID());
     }
 
-    /**
-     * @return int
-     */
     public function getGID(): int
     {
         return $this->gID;
     }
 
     /**
-     * @param int $gID
-     * @return GroupSignup
+     * @return $this
      */
-    public function setGID(int $gID): GroupSignup
+    public function setGID(int $gID): self
```

Comment:
There was a bug here: this method returns the instance itself (`GroupRoleChange`), not an instance of `GroupSignup`